src/execution/values.js

```diff
@@ -191,7 +191,7 @@ function coerceValue(type: GraphQLInputType, value: mixed): mixed {
     const itemType = type.ofType;
     if (isCollection(_value)) {
       const coercedValues = [];
-      const valueIter = createIterator(_value);
+      const valueIter = createIterator((_value: any));
```

Comment:
Could you remove this line and rebase this PR from master branch again?

Some of the less-related parts of this PR have been already merged into master